orm/serial_bucket.go

```diff
@@ -258,6 +258,7 @@ func (smb *serialModelBucket) IndexScan(db weave.ReadOnlyKVStore, indexName stri
 }
 
 func (smb *serialModelBucket) ByIndex(db weave.ReadOnlyKVStore, indexName string, key []byte, destination SerialModelSlicePtr) error {
+	// TODO refactor this function to return an iterator
```

Comment:
Please don't leave _TODO_ in code. Instead, explain **why** it is implemented like it is and how could it be done better.

Leaving `TODO do X` never motivates to address this comment. As a result, this comment will stay here forever providing no information beyond an impression that this code is half done.